File: import_csv.js
const sqlite3 = require('sqlite3').verbose();
const fs = require('fs');
const path = require('path');

const DB_NAME = 'contacts.db';
const CSV_FILE = 'contacts.csv';

const db = new sqlite3.Database(DB_NAME);

db.run(`CREATE TABLE IF NOT EXISTS contacts (
  id INTEGER PRIMARY KEY AUTOINCREMENT,
  name TEXT NOT NULL,
  email TEXT,
  phone TEXT,
  city TEXT,
  "group" TEXT
)`);

const content = fs.readFileSync(CSV_FILE, 'utf-8');
const lines = content.split(/\r?\n/);

const stmt = db.prepare(`INSERT INTO contacts (name, email, phone, city, "group") VALUES (?, ?, ?, ?, ?)`);

for (let i = 0; i < lines.length; i++) {
  const row = lines[i].trim();
  if (row === '') continue;
  const cols = row.split(',');
  if (cols.length >= 3) {
    const name = cols[0].trim();
    const email = cols[1].trim();
    const phone = cols[2].trim();
    const city = "";
    const group = "سایر";
    stmt.run(name, email, phone, city, group);
  }
}

stmt.finalize(() => {
  console.log('✅ داده‌های contacts.csv با موفقیت به دیتابیس اضافه شدند.');
  db.close();
});

File: import_csv.py
import sqlite3
import csv

DB_NAME = "contacts.db"

def import_csv():
    conn = sqlite3.connect(DB_NAME)
    c = conn.cursor()
    
    try:
        with open('contacts.csv', 'r', encoding='utf-8') as f:
            reader = csv.reader(f)
            for row in reader:
                if len(row) >= 3:
                    name = row[0].strip()
                    email = row[1].strip()
                    phone = row[2].strip()
                    city = ""          
                    group = "سایر"     
                    
                    c.execute('''
                        INSERT INTO contacts (name, email, phone, city, group)
                        VALUES (?, ?, ?, ?, ?)
                    ''', (name, email, phone, city, group))
        conn.commit()
        print(" داده‌های فایل contacts.csv با موفقیت به دیتابیس اضافه شدند.")
    except FileNotFoundError:
        print(" فایل contacts.csv پیدا نشد. لطفاً فایل را در مسیر صحیح قرار دهید.")
    except Exception as e:
        print(f" خطا: {e}")
    finally:
        conn.close()

if __name__ == "__main__":
    import_csv()
